### PR TITLE
Fixes #1759 - Refactor multiselect

### LINKED ISSUE #1759: Refactor ClayMultiselect

### PR DESCRIPTION
**Pull request by carloslancha**
EVENTS CHANGED:
buttonClicked -> selectButtonClick
inputOnBlur -> inputBlur
itemAdded -> labelItemAdded
itemFocused -> labelItemFocused
itemRemoved -> labelItemRemoved

EVENTS ADDED:
autocompleteItemSelected
inputFocus
labelItemClick
labelItemCloseButtonClick
labelItemKeyDown

EVENTS UNTOUCHED:
dataChange
inputChange

API CHANGED
allowedCharacters -> inputAllowedCharacters
debounceTime -> dataProviderDebounceTime
extractData -> autocompleteFilterCondition
initialData -> dataProviderInitialData

BEHAVIOR CHANGES
Arrow keys must do nothing

### CODE REVIEW

**Comment by matuzalemsteles** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: packages/clay-multi-select/src/ClayMultiSelect.js [hunk 1]
```diff
@@ -171,287 +207,298 @@ class ClayMultiSelect extends ClayComponent {
...
+	_addLabelItem(item) {
+		const {label, value} = item;
 
-		if (event.data.char === ',' || sentence.length > 1) {
-			sentence.forEach(value => {
-				this._handleItemAdded(
-					value,
-					this._getItemSchema(value, value),
-					event
+		const filteredItem = this.filteredItems
```
Comment (matuzalemsteles) [inline packages/clay-multi-select/src/ClayMultiSelect.js]: This is good hack to dribble what was typed

File: packages/clay-alert/package.json [hunk 2]
```diff
@@ -21,7 +21,8 @@
     "compile": "babel -d lib/ src/ -s --ignore src/__tests__",
     "link": "npm link",
     "prepublish": "npm run soy && npm run compile",
-    "soy": "metalsoy --soyDeps '../../node_modules/clay-+(button|icon)/src/**/*.soy'"
+    "soy": "metalsoy --soyDeps '../../node_modules/clay-+(button|icon)/src/**/*.soy'",
+    "yarnLink": "yarn link"
```
Comment (wincent) [inline packages/clay-alert/package.json]: How about instead of adding `yarnLink` we just change `"link": "npm link"` to `"link": "yarn link"`?

Comment (carloslancha) [reply packages/clay-alert/package.json]: If nobody wants the option of linking with npm anymore I'm ok with that. @jbalsas ?

Comment (jbalsas) [reply packages/clay-alert/package.json]: we're killing `npm` usage for now, so yeah, just `link: yarn link` sounds fine

**Comment by carloslancha** [comment]
Also @matuzalemsteles I didn't answer you:

- Regarding the MultiSelect focus, I think we should keep focus on the input even when focusing on the label: From the point of view of UX this is not the expected behavior, from the point of view of code I find that pretty confusing.

Why are we removing label interactions?: Talked with UX and @marcoscv and they told me that this interactions shouldn't happen with arrows but tab.

**Comment by matuzalemsteles** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by carloslancha** [comment]
Hey @matuzalemsteles we need to relase this ASAP, some teams are blocked because of this. If this is good enough to go with it we can take care of the little things after it.

File: packages/clay-multi-select/src/ClayMultiSelect.js [hunk 3]
```diff
@@ -171,287 +207,298 @@ class ClayMultiSelect extends ClayComponent {
...
 
-			if (invalidWord) {
-				this.inputValue = invalidWord;
-			}
-		} else {
-			this.inputValue = invalidWord;
+		item = filteredItem || item;
+
+		if (value && !this._isItemSelected(item)) {
+			this.selectedItems.push(item);
```
[the remark this answers is not in the document]
Comment (matuzalemsteles) [reply packages/clay-multi-select/src/ClayMultiSelect.js]: I did not understand what to keep `selectedItems` synced would solve this case, but I also think we should do this. If we do not include this, we will have the commas being added to the input hidden that are causing this problem https://issues.liferay.com/browse/LPS-91089.

Or will maintaining sync solve this? I did not test it yet.

**Pull request merged**